src/Microsoft.Identity.Client/User.cs

```diff
@@ -47,37 +46,34 @@ internal User(User other)
             IdentityProvider = other.IdentityProvider;
         }
 
+        public User(string identifier, string displayableId, string name, string identityProvider)
+        {
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            DisplayableId = displayableId;
+            Name = name;
+            IdentityProvider = identityProvider;
```

Comment:
These changes are not about unit testing MSAL or getting coverage of the MSAL API - they are about allow the developer to test their own application.
e.g. the application developer might use a property like _User.DisplayableId_ in their app, so they need to be able to create a dummy user that has known properties so they can test their app does the right thing with it.
They might also make a sequence of calls to MSAL - call (1) returns a user object that is later used in call (2). If they want to test their app passes the _User_ object correctly between calls then they need to be able to write a unit test that creates a dummy _User_ object.